src/zenml/integrations/vault/__init__.py

```diff
@@ -0,0 +1,53 @@
+#  Copyright (c) ZenML GmbH 2021. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Initialization for the Vault Secret Manager integration.
+
+The Vault secrets manager integration submodule provides a way 
+to access the HashiCorp vault secrets manager from within you ZenML 
+Pipeline runs.
+"""
+from typing import List
+
+from zenml.enums import StackComponentType
+from zenml.integrations.constants import VAULT
+from zenml.integrations.integration import Integration
+from zenml.zen_stores.models import FlavorWrapper
+
+VAULT_SECRETS_MANAGER_FLAVOR = "vault"
+
+
+class VaultSecretManagerIntegration(Integration):
+    """Definition of HashiCorp Vault integration with ZenML."""
+
+    NAME = VAULT
+    REQUIREMENTS = ["hvac>=0.11.2"]
+
+    @classmethod
+    def flavors(cls) -> List[FlavorWrapper]:
+        """Declare the stack component flavors for the Vault integration.
+
+        Returns:
+            List[FlavorWrapper]: The list of stack component flavors.
+        """
+        return [
+            FlavorWrapper(
+                name=VAULT_SECRETS_MANAGER_FLAVOR,
+                source="zenml.integrations.vault.secrets_manager.VaultSecretsManager",
```

Comment:
Also noting that in `src/zenml/integrations/__init__.py` it uses `VaultSecretManager` (singular) but here it uses `VaultSecretsManager` (plural)... This one doesn't get checked, so maybe that's why you didn't see it. In any case, it should be plural.